fuel-core/src/database/contracts.rs

```diff
@@ -49,6 +65,23 @@ impl Storage<ContractId, UtxoId> for Database {
     }
 }
 
+impl Database {
+    pub fn contract_balances(
+        &self,
+        contract: ContractId,
+        start_asset: Option<AssetId>,
+        direction: Option<IterDirection>,
+    ) -> impl Iterator<Item = Result<AssetId, Error>> + '_ {
+        self.iter_all::<Vec<u8>, Word>(
+            BALANCES,
+            Some(contract.as_ref().to_vec()),
+            start_asset.map(|b| contract_asset_id_id_key(&contract, &b)),
```

Comment:
[nit] consider reusing the same compound key wrapper type used for fetching and inserting balances in other places:

```suggestion
            start_asset.map(|b| MultiKey::new((&contract, &b)).as_ref().to_vec()),
```

Reply:
I had seen this method of conversion somewhere else, but `MultiKey` is much cleaner. I committed this but had to do it myself because this also means the removal of 2 functions in the same file and the addition of an import